rust-runtime/aws-smithy-http-server/src/extension.rs

```diff
@@ -34,10 +34,51 @@
 
 //! Extension extraction to share state across handlers.
 
-use super::rejection::{ExtensionRejection, ExtensionsAlreadyExtracted, MissingExtension};
+use super::rejection::{ExtensionHandlingRejection, ExtensionsAlreadyExtracted, MissingExtension};
 use async_trait::async_trait;
 use axum_core::extract::{FromRequest, RequestParts};
-use std::ops::Deref;
+use std::{borrow::Cow, ops::Deref};
+
+/// Extension type used to store the Smithy model namespace.
+#[derive(Debug, Clone)]
+pub struct ExtensionNamespace(pub Cow<'static, str>);
```

Comment:
If we have `new()` methods, we can make the wrapped type private in all of these.

Reply:
Huh, right, let me change.